src/EmbedIO/Net/Internal/HttpListenerRequest.cs

```diff
@@ -304,8 +303,8 @@ internal void FinishInitialization()
             }
 
             CreateQueryString(_url.Query);
-
-            if (!_clSet && (HttpVerb == HttpVerbs.Post || HttpVerb == HttpVerbs.Put))
+            
+            if (ContentLength64 < 0 && (HttpVerb == HttpVerbs.Post || HttpVerb == HttpVerbs.Put))
```

Comment:
Shouldn't be `ContentLength64 == 0` or `<=`? @geoperez

Reply:
Done!